### ISSUE TITLE
systemd-backlight saves value 0 when notebook lid is closed on service stop

### ISSUE THREAD

**Issue by VeldoraTheDragon** [systemd/systemd#29328]
### systemd version the issue has been seen with

systemd-254.3-1-x86_64

### Used distribution

Arch Linux

### Linux kernel version used

6.4.12-arch1-1

### CPU architectures issue was seen on

x86_64

### Component

other

### Expected behaviour you didn't see

When the notebook-lid is closed while attached to an external screen and the service `systemd-backlight@backlight:intel_backlight` get's stopped, the value that's saved in `/sys/class/backlight/intel_backlight/brightness` should get saved into `/var/lib/systemd/backlight/pci-0000:00:02.0:backlight:intel_backlight`.

So when starting the service, the correct value can be loaded.

### Unexpected behaviour you saw

When the lid is closed, instead the value 0 gets saved into `/var/lib/systemd/backlight/pci-0000:00:02.0:backlight:intel_backlight` which ends up in correcting the value to 5% of `max_brightness` on service start.

### Steps to reproduce the problem

To test the behaviour the following steps have been executed:

* Connect to target machine via SSH
* Increase Display Brightness on notebook to at least 50%.
* Check value with `cat /sys/class/backlight/intel_backlight/brightness`
* Close the LID
* Save the value by stopping the service: `systemctl stop systemd-backlight@backlight:intel_backlight`
* Check the saved value with `cat /var/lib/systemd/backlight/pci-0000\:00\:02.0\:backlight\:intel_backlight`
* Open the LID
* Load the saved value by starting the service: `systemctl start systemd-backlight@backlight:intel_backlight`
* Output: `systemd-backlight[17552]: intel_backlight: Saved brightness 0 is too low; increasing to 969.`

**Comment by poettering** [comment]
This smells like a driver bug. We just save whatever the kernel driver reports to us.

**Comment by yuwata** [comment]
@VeldoraTheDragon 
> How does the code retrieve the value? Is there something similar as a cli, so I can test it manually?

Basically, systemd-backlight retrieves and determine brightness from the combination of
/sys/class/backlight/intel_backlight/actual_brightness
/sys/class/backlight/intel_backlight/brightness
/sys/class/backlight/intel_backlight/max_brightness
See read_brightness() in src/backlight/backlight.c for more details.
https://github.com/systemd/systemd/blob/12696a7dbabb1271cbfbaa035c8403ffd62f0789/src/backlight/backlight.c#L386-L443

**Comment by yuwata** [comment]
Please try to enable to generate debugging logs.
Creating the following drop-in config by `sudo systemctl edit systemd-backlight@.service` makes the service generate debugging logs.
```
# /etc/systemd/systemd-backlight@.service
[Service]
Environment=SYSTEMD_LOG_LEVEL=debug
```

**Comment by YHNdnzj** [comment]
> And after all things fail. (file does not exist or value is below 0) it fallbacks to `/sys/class/backlight/intel_backlight/brightness`, which in my eyes would be the better procedure for notebooks attached to external screens.

The behavior was changed in https://github.com/systemd/systemd/issues/16302 because for NVIDIA drivers `brightness` doesn't work correctly... Maybe we should make this overrideable.

**Comment by yuwata** [comment]
Introducing an env variable to prioritize brightness rather than actual_brightness may be good for such case, e.g. `$SYSTEMD_BACKLIGHT_USE_BRIGHTNESS` or `$SYSTEMD_BACKLIGHT_IGNORE_ACTUAL_BRIGHTNESS`.

**Comment by yuwata** [comment]
@VeldoraTheDragon Could you provide the values of `brightness` and `actual_brightness` on several situations, e.g. using laptop display, lid is closed with an external display connected, lid is closed without an external display, and so on.

**Comment by yuwata** [comment]
> Another option would be, if the "correction" value could be overrideable. Instead by default 5% of the `max_brightness`, it could be configured to e.g. 50% by using an environment-variable.

I noticed we have already introduced to support `ID_BACKLIGHT_CLAMP` udev boolean property. It may be better to make the property also take a percentage.

**Comment by yuwata** [comment]
Fix is waiting in #29443. If you'd like, please test the PR.
